common/src/Microsoft.Azure.IIoT.Agent.Framework/src/Storage/Database/Services/JobDatabase.cs

```diff
@@ -27,31 +27,46 @@ public class JobDatabase : IJobRepository {
             var dbs = databaseServer.OpenAsync(databaseJobRepositoryConfig.DatabaseName).Result;
             var cont = dbs.OpenContainerAsync(databaseJobRepositoryConfig.ContainerName).Result;
             _documents = cont.AsDocuments();
+            _lock = new SemaphoreSlim(1, 1);
         }
 
         /// <inheritdoc/>
         public async Task<JobInfoModel> AddAsync(JobInfoModel job, CancellationToken ct) {
             if (job == null) {
                 throw new ArgumentNullException(nameof(job));
             }
-            while (true) {
-                var document = await _documents.FindAsync<JobDocument>(job.Id, ct);
-                if (document != null) {
-                    throw new ConflictingResourceException($"Job {job.Id} already exists.");
-                }
-                job.LifetimeData.Created = job.LifetimeData.Updated = DateTime.UtcNow;
+            var retries = 0;
+            var exceptions = new List<Exception>();
+            while (retries < MaxRetries) {
+                await _lock.WaitAsync().ConfigureAwait(false);
```

Comment:
Let's pass cancellation token to all calls to `_lock.WaitAsync()` in `JobDatabase.cs`.